dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java

```diff
@@ -453,6 +501,40 @@ public void updateIndex(Context context, boolean force) {
         }
     }
 
+    private void startMultiThreadIndex(Context context, boolean force, List<UUID> ids) throws SQLException {
+        int numThreads = ConfigurationManager.getIntProperty("discovery", "indexer.items.threads", 5);
+
+        if (ids == null) {
+            ids = new ArrayList<>();
+            Iterator<Item> items = itemService.findAllUnfiltered(context);
+            for (Item item : ImmutableList.copyOf(items)) {
+                ids.add(item.getID());
+            }
+
+            for (WorkspaceItem wsi : workspaceItemService.findAll(context)) {
+                ids.add(wsi.getItem().getID());
+            }
+
+            for (WorkflowItem wfi : workflowItemService.findAll(context)) {
+                ids.add(wfi.getItem().getID());
+            }
+        }
+        List<UUID>[] arrayIDList = Util.splitList(ids, numThreads);
+        List<IndexerThread> threads = new ArrayList<IndexerThread>();
+        for (List<UUID> hl : arrayIDList) {
+            IndexerThread thread = new IndexerThread(hl, force);
+            thread.start();
+            threads.add(thread);
+        }
+        boolean finished = false;
+        while (!finished) {
```

Comment:
USe the spring TaskExecutors to find a better way to handle this.

Reply:
support for multi-threads has been withdrawn from this PR as suggested by @cwilper https://github.com/4Science/DSpace/pull/65#issuecomment-477177919 and will be eventually introduced in a separate PR/ticket